analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -111,54 +107,37 @@ private void writeMarkerFile(File rootDir) throws IOException {
     private Collection<ProjectVersionRef> getAllProjectDependencies(Project project) {
         final Set<ProjectVersionRef> result = new LinkedHashSet<>();
         project.getConfigurations().all(configuration -> {
-            final ResolutionResult resolutionResult = configuration.getIncoming()
-                    .getResolutionResult();//force dependency resolution - this is needed later on if we encounter deps with no version
-            configuration.getAllDependencies().forEach(dep -> {
-                if (dep instanceof DefaultSelfResolvingDependency) {
-                    logger.warn("Ignoring dependency of type {} on project {}", dep.getClass().getName(), project.getName());
-                } else if (isEmpty(dep.getVersion())) {
-                    // look up resolved dependency with a toString hack
-                    boolean found = false;
-                    final Set<? extends DependencyResult> resolvedDependencies = getAllDependenciesFromResolutionResult(
-                            resolutionResult);
-                    for (DependencyResult resolvedDependency : resolvedDependencies) {
-                        if (!(resolvedDependency instanceof DefaultResolvedDependencyResult)) {
-                            continue;
-                        }
 
-                        final String resolvedDependencyDisplayName = ((DefaultResolvedDependencyResult) resolvedDependency)
-                                .getSelected()
-                                .getId().getDisplayName();
-                        // the display name should be something like: "org.example:somelib:1.2.3"
-                        if (resolvedDependencyDisplayName.startsWith(dep.getGroup() + ":" + dep.getName()) && StringUtils
-                                .countMatches(resolvedDependencyDisplayName, ":") == 2) {
-                            final int i = resolvedDependencyDisplayName.lastIndexOf(":");
-                            if (i != -1) {
-                                result.add(ProjectVersionFactory.withGAVAndConfiguration(dep.getGroup(), dep.getName(),
-                                        resolvedDependencyDisplayName.substring(i + 1),
-                                        configuration.getName()));
-                                found = true;
-                            }
+            if (configuration.isCanBeResolved()) {
+
+                LenientConfiguration lenient = configuration.getResolvedConfiguration().getLenientConfiguration();
+
+                if (lenient.getUnresolvedModuleDependencies().size() > 0) {
```

Comment:
Not sure if this should be an exception - it means things like `org.foo:foobar:9.9.9` wasn't a valid dependency and couldn't be resolved.

Reply:
I think it should be

Reply:
@geoand I've added configuration for it ( by default it will throw ). This will allow for any potential corner cases if a project has a midstream with -rh-n increment values which doesn't resolve in repour.